File: examples/node/randomLetter.js
// Pick a random letter from a language's alphabet.
const { getLowercase, getUppercase } = require('../..');

async function main() {
  const code = process.argv[2] || 'en';
  const script = process.argv[3];
  let letters = await getLowercase(code, script);
  if (!letters.length) {
    letters = await getUppercase(code, script);
  }
  const pick = letters[Math.floor(Math.random() * letters.length)];
  console.log(pick);
}

main();

File: examples/python/random_letter.py
"""Pick a random letter from a language's alphabet."""
from __future__ import annotations

import random
import sys

from worldalphabets import load_alphabet


def main() -> None:
    code = sys.argv[1] if len(sys.argv) > 1 else "en"
    script = sys.argv[2] if len(sys.argv) > 2 else None
    alphabet = load_alphabet(code, script=script)
    letters = alphabet.lowercase or alphabet.alphabetical
    print(random.choice(letters))


if __name__ == "__main__":
    main()
